doc/dev-guide/src/release-process.md

```diff
@@ -1,66 +1,217 @@
-# Making a release
+# Release Process
 
-Before making a release, ensure that `rustup-init.sh` is behaving correctly,
+This document describes the process for making a new release of Rustup. It is
+split into the two sections. The first section explains the release process,
+while the second section documents for maintainers how to make a release.
```

Comment:
"documents how to make a release"? The "for maintainers" part reads a little awkwardly.

(Also "make" feels a little vague here, can we make that more precise?)

Reply:
I've rewritten the into a bit, but I'm not sure how much better it is now. 😅